### PR TITLE
Marine B-matrix refactor companion PR

### PR DESCRIPTION
**Pull request by DavidNew-NOAA**
This PR is a companion to Global Workflow PR [#2992](https://github.com/NOAA-EMC/global-workflow/pull/2992) and JCB-GDAS PR [#36](https://github.com/NOAA-EMC/jcb-gdas/pull/36).

It makes a few necessary changes to the marine JCB base YAML required to have JCB working in the B-matrix task in the Global Workflow.

### CODE REVIEW

**Review by danholdaway — DISMISSED** [review]

**Review by guillaumevernieres — DISMISSED** [review]

File: parm/soca/soca_fix_stage_500.yaml.j2
Comment (AndrewEichmann-NOAA) [inline parm/soca/soca_fix_stage_500.yaml.j2:12]: Is this functionality being replaced somewhere? Because the yet-unmerged LETKF uses it.

Comment (guillaumevernieres) [reply parm/soca/soca_fix_stage_500.yaml.j2:12]: @AndrewEichmann-NOAA , I missed it when reviewing then. You should not use ` fields_metadata.yaml` from the fix_dir. We need to have that file removed at some point.

Comment (DavidNew-NOAA) [reply parm/soca/soca_fix_stage_500.yaml.j2:12]: @AndrewEichmann-NOAA Like @guillaumevernieres said. The current ```fields_metadata.yaml``` is actually staged here https://github.com/NOAA-EMC/GDASApp/blob/34f894ffc8ff15b1a43d537a0a1043670231e5a3/parm/soca/soca_utils_stage.yaml.j2#L5

**Review by guillaumevernieres — DISMISSED** [review]

**Comment by DavidNew-NOAA** [comment]
@guillaumevernieres @AndrewEichmann-NOAA @danholdaway @CoryMartin-NOAA Can I get a final review on this PR, so we can merge and hopefully move forward on GW PR [#2992](https://github.com/NOAA-EMC/global-workflow/pull/2992) next week?

**Review by CoryMartin-NOAA — APPROVED** [review]
Looks good to me, merge once @guillaumevernieres is happy

**Review by danholdaway — APPROVED** [review]

**Review by guillaumevernieres — APPROVED** [review]

**Comment by DavidNew-NOAA** [comment]
This PR will break marine testing until GW PR [#2992](https://github.com/NOAA-EMC/global-workflow/pull/2992) is merged, so I might as well wait until I meet with Rahul and Walter next week to hash out the issues with that PR before I merge this one. The companion jcb-gdas PR that just got merged shouldn't break anything though, so that's good.

**Comment by guillaumevernieres** [comment]
> This PR will break marine testing until GW PR [#2992](https://github.com/NOAA-EMC/global-workflow/pull/2992) is merged, so I might as well wait until I meet with Rahul and Walter next week to hash out the issues with that PR before I merge this one. The companion jcb-gdas PR that just got merged shouldn't break anything though, so that's good.

develop is broken anyways, the default state ... But yes, lets hold on!

**Review by AndrewEichmann-NOAA — APPROVED** [review]
Looks good

**Comment by DavidNew-NOAA** [comment]
I'm going to go ahead and merge this PR since GW PR [#2992](https://github.com/NOAA-EMC/global-workflow/pull/2992) should be ready for review soon after discussing with the GW team yesterday. Let me know if anyone objects.

**Comment by guillaumevernieres** [comment]
> I'm going to go ahead and merge this PR since GW PR [#2992](https://github.com/NOAA-EMC/global-workflow/pull/2992) should be ready for review soon after discussing with the GW team yesterday. Let me know if anyone objects.

Go for it. We'll have bugfixes coming in after yours (https://github.com/NOAA-EMC/GDASApp/pull/1359), but what you have in this PR should still be OK with the low-res test cases.

**Pull request merged**
